### DISCUSSION TITLE
ZIgBee

### DISCUSSION THREAD

**Discussion by storchalberto** [arendst/Tasmota#21214] [Support request]
I want to interface a ZiGbee EFR32MG2X board to my esp32. I have configured GPIO5 as Zigbee TX and connected to PB00 - RX of the modue, and the and GPIO35 as Zigbee RX conneced to PB01 - TX of the module.
The button for operating the zigbee are disabled, on the log i see: 

00:00:00.593 ZIG: GPIOs Rx:35 Tx:5
00:00:00.595 ZIG: Serial UART2
.
.
.
16:16:57.014 ZIG: rebooting ZNP device
16:17:02.202 ZIG: timeout, goto label 99
16:17:02.204 RSL: RESULT = {"ZbState":{"Status":99,"Message":"Abort"}}
16:17:02.210 ZIG: Abort
The module is indicated as:
Firmware NCP EZSP-V8
Version V 6.10.3
Any suggestion?

**Comment by sfromis** [comment]
You are using the wrong binary, tasmota32-zbbrdgpro.bin won't work for that chip, better use the "unofficial" `tasmota32-zigbeebridge` from https://github.com/tasmota/install/blob/firmware/firmware/unofficial/

Also, the existing Tasmota config for a "similar" device using ESP32 and EFR32 is also configuring the `Zigbee Rst` pin.
https://templates.blakadder.com/ewelink_ZB-GW03.html

And you also need to have the suitable Zigbee firmware flashed to the chip.

**Reply by storchalberto** [reply]
I do not want to develop a standard zigbee bridge, i want to interface sensors to my berry software to develop a custom application. In a standard configuration of tasmota sw, this is supported?. Fot the zigbee firmware on the chip, the
Firmware NCP EZSP-V8
Version V 6.10.3
is ok?

**Reply by storchalberto** [reply]
connected GPIO02 as Zigbee RST 1, I have downloaded from the "unofficial" repository the tasmota32-zigbee.bin, here the log
18:55:33.195 ZIG: rebooting ZNP device
18:55:33.268 ZbInput discarding byte 01
18:55:33.268 ZbInput discarding byte 00
18:55:33.269 ZbInput discarding byte 00
18:55:33.270 ZbInput discarding byte 00
18:55:33.281 ZbInput discarding byte 00
18:55:33.281 ZbInput discarding byte F8
18:55:33.282 ZbInput discarding byte 00
18:55:33.283 ZbInput discarding byte 00
18:55:33.294 ZbInput discarding byte 00
18:55:33.294 ZbInput discarding byte 10
18:55:33.295 ZbZNPReceived: received bad FCS frame FE0100000000, 1
18:55:33.378 ZbInput discarding byte C0
18:55:33.379 ZbInput discarding byte 00
18:55:33.379 ZbInput discarding byte 00
18:55:33.380 ZbInput discarding byte 00
18:55:33.391 ZbInput discarding byte 00
18:55:33.392 ZbInput discarding byte E0
18:55:33.392 ZbInput discarding byte 01
18:55:33.393 ZbInput discarding byte 00
18:55:33.404 ZbInput discarding byte 00
18:55:33.405 ZbInput discarding byte 00
18:55:33.405 ZbInput discarding byte C0
18:55:33.416 ZbInput discarding byte 05
18:55:33.417 ZbInput discarding byte 00
18:55:33.418 ZbInput discarding byte 00
18:55:33.419 ZbInput discarding byte 00
18:55:33.430 ZbInput discarding byte 00
18:55:33.430 ZbInput discarding byte FC
18:55:33.431 ZbInput discarding byte 01
18:55:33.432 ZbInput discarding byte 00
18:55:33.443 ZbInput discarding byte 00
18:55:33.444 ZbInput discarding byte 00
18:55:33.444 ZbInput discarding byte F8
18:55:33.455 ZbInput discarding byte 00
18:55:33.456 ZbInput discarding byte 00
18:55:33.457 ZbInput discarding byte 00
18:55:33.458 ZbInput discarding byte 00
18:55:33.469 ZbInput discarding byte 00
18:55:33.470 ZbInput discarding byte FD
18:55:33.470 ZbInput discarding byte 01
18:55:33.471 ZbInput discarding byte 00
18:55:33.482 ZbInput discarding byte 00
18:55:33.483 ZbInput discarding byte 00
18:55:33.484 ZbInput discarding byte 00
18:55:33.495 ZbInput discarding byte F8
18:55:33.496 ZbInput discarding byte 00
18:55:33.497 ZbInput discarding byte 00
18:55:33.498 ZbInput discarding byte 00
18:55:33.509 ZbInput discarding byte 00
18:55:33.509 ZbInput discarding byte FC
18:55:33.510 ZbInput discarding byte 01
18:55:33.511 ZbInput discarding byte 00
18:55:33.522 ZbInput discarding byte 00
18:55:33.523 ZbInput discarding byte 00
18:55:33.524 ZbInput discarding byte FB
18:55:33.525 ZbInput discarding byte 00
18:55:33.536 ZbInput discarding byte 00
18:55:33.537 ZbInput discarding byte 00
18:55:33.538 ZbInput discarding byte 11
18:55:34.549 ZbInput discarding byte 1A
18:55:34.550 ZbInput discarding byte C1
18:55:34.551 ZbInput discarding byte 02
18:55:34.552 ZbInput discarding byte 02
18:55:34.563 ZbInput discarding byte 9B
18:55:34.564 ZbInput discarding byte 7B
18:55:34.565 ZbInput discarding byte 7E
18:55:38.326 ZIG: timeout, goto label 99
18:55:38.328 RSL: RESULT = {"ZbState":{"Status":99,"Message":"Abort"}}
18:55:38.332 ZIG: Abort
18:55:38.333 ZIG: Stopping (99)

**Reply by sfromis** [reply]
**Again**, you're using the wrong Tasmota build. Use the one I suggested. There are different builds for ZNP and EZSP.

The only role Tasmota supports with Zigbee is to act as a coordinator. With ESP32, this includes the ability to control it via Berry on the coordinator itself, instead of depending on a backend for that.

And I'd expect it to be a requirement to have flashed suitable firmware. I've not worked directly with the EFR32, but when flashed to be working, Tasmota reports:
`{"ZbState":{"Status":55,"Version":"6.7.9.0","Protocol":8,"Stack":2}}`
IDK if yours would work or not.

**Reply by storchalberto** [reply]
I have tried to compile using benzino-developement and enabling the EZSP protocol. i see this log:
19:39:31.554 RSL: RESULT = {"ZbState":{"Status":1,"Message":"EFR32 EZSP booted","RestartReason":"Power-on","Code":2}}
19:39:31.665 RSL: RESULT = {"ZbState":{"Status":55,"Version":"6.10.3.0","Protocol":8,"Stack":2}}
19:39:31.682 RSL: RESULT = {"ZbState":{"Status":3,"Message":"Configured, starting coordinator"}}
19:39:35.624 RSL: RESULT = {"ZbState":{"Status":56,"IEEEAddr":"0x28DBA7FFFE501034","ShortAddr":"0x0000","DeviceType":1}}
19:39:35.930 ZIG: Subscribe to group 0 'ZbListen0 0'
19:39:36.113 RSL: RESULT = {"ZbState":{"Status":0,"Message":"Started"}}
19:39:36.129 ZIG: Zigbee started
19:39:36.139 ZIG: No Zigbee device information
19:39:36.145 ZIG: No Zigbee device data
19:39:36.148 ZIG: ZbLoad '<internal_plugin>' loaded successfully
19:39:44.722 CMD: restart
19:39:44.726 RSL: RESULT = {"Restart":"1 to restart, 2 to halt"}
So the version is 6.10.3.0. 
It is the problem?
if so, were i can download the correct version and how i can upload the module?

**Reply by sfromis** [reply]
What you show now looks like it's working, basically waiting for you to pair a Zigbee device.

**Reply by storchalberto** [reply]
Thanks, now the zigbee buttons are enabled, i will try to pair a device (i am waiting for receiving a door contact zigbee)
